### PR TITLE
KAFKA-16448: Add ProcessingExceptionHandler in Streams configuration

### PR DESCRIPTION
**Pull request by loicgreffier**
This PR is part of KAFKA-16448 which aims to bring a ProcessingExceptionHandler to Kafka Streams in order to deal with exceptions that occur during processing.

This PR brings `ProcessingExceptionHandler` in Streams configuration.

Jira: https://issues.apache.org/jira/browse/KAFKA-16448.

### Contributors

@Dabz 
@sebastienviale
@loicgreffier

### Depends On

https://github.com/apache/kafka/pull/16090

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -926,6 +933,11 @@ public class StreamsConfig extends AbstractConfig {
                     DefaultProductionExceptionHandler.class.getName(),
                     Importance.MEDIUM,
                     DEFAULT_PRODUCTION_EXCEPTION_HANDLER_CLASS_DOC)
+            .define(PROCESSING_EXCEPTION_HANDLER_CLASS_CONFIG,
+                    Type.CLASS,
+                    LogAndFailProcessingExceptionHandler.class.getName(),
+                    Importance.MEDIUM,
+                    PROCESSING_EXCEPTION_HANDLER_CLASS_DOC)
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:945]: Could you please add unit tests in `StreamsConfigTest`?
I think you need the following unit tests:
- One where config is not set to verify the default value
- One where the config is set and you get the correct instance from `processingExceptionHandler()`.

**Review by cadonna — COMMENTED** [review]
@loicgreffier Thanks for the PR!

Here my comment.

Since today is feature freeze, it would be great if you could make the changes as soon as possible.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (sebastienviale) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:945]: hi, 
It is done !

File: streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java [hunk 2]
```diff
@@ -1579,6 +1579,27 @@ public void shouldDisableMetricCollectionOnMainConsumerOnly() {
...
+        props.put(StreamsConfig.PROCESSING_EXCEPTION_HANDLER_CLASS_CONFIG, "org.apache.kafka.streams.errors.LogAndContinueProcessingExceptionHandler");
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+
+        assertEquals("org.apache.kafka.streams.errors.LogAndContinueProcessingExceptionHandler",   streamsConfig.processingExceptionHandler().getClass().getName());
+    }
+
+    @Test
+    public void testInvalidProcessingExceptionHandler() {
+        props.put(StreamsConfig.PROCESSING_EXCEPTION_HANDLER_CLASS_CONFIG, "org.apache.kafka.streams.errors.InvalidProcessingExceptionHandler");
+        assertThrows(ConfigException.class, () -> new StreamsConfig(props));
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java]: Could you please also verify the message?
```suggestion
        final Exception exception = assertThrows(ConfigException.class, () -> new StreamsConfig(props));
        assertThat(
            exception.getMessage(),
            containsString("<the message to verify>")
        );
```

**Review by cadonna — COMMENTED** [review]
Thanks for the updates, @loicgreffier and @sebastienviale!

After addressing my comment, I will approve and merge this PR.

File: streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java (hunk 2 above)
Comment (sebastienviale) [reply streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java]: Verification has been added

**Review by cadonna — APPROVED** [review]
Thanks @loicgreffier @sebastienviale !

LGTM!

**Comment by loicgreffier** [comment]
@cadonna I've rebased this PR so file changes are now clear. Do we need to squash commits manually or is it automatically done when merging ?

**Comment by cadonna** [comment]
> @cadonna I've rebased this PR so file changes are now clear. Do we need to squash commits manually or is it automatically done when merging ?

Thanks! Squashing happens when merging. Don't worry!

**Pull request merged**